File: Backend/BaseMgtService/Senior.AgileAI.BaseMgt.Application/DTOs/AddReqManuallyDTO.cs
using Senior.AgileAI.BaseMgt.Domain.Enums;

#nullable disable

namespace Senior.AgileAI.BaseMgt.Application.DTOs
{
    public class AddReqManuallyDTO
    {
        public Guid ProjectId { get; set; }
        public List<ReqDTO> Requirements { get; set; }
    }

    public class ReqDTO
    {
        public required string Title { get; set; }
        public required ReqPriority Priority { get; set; }
        public required RequirementsStatus Status { get; set; }
        public required string Description { get; set; }

    }
}

File: agilemeets_mobile/lib/data/models/requirements/add_req_manually_dto.dart
import 'req_dto.dart';

class AddReqManuallyDTO {
  final String projectId;
  final List<ReqDTO> requirements;

  const AddReqManuallyDTO({
    required this.projectId,
    required this.requirements,
  });

  Map<String, dynamic> toJson() => {
    'projectId': projectId,
    'requirements': requirements.map((e) => e.toJson()).toList(),
  };
} 